### ISSUE TITLE
Similator is not found, Xcode and simulator installed and working properly.

### ISSUE THREAD

**Issue by unutoiul** [facebook/react-native#23878]
Starting Metro Bundler on port 19004.
Simulator is installed but is identified as 'com.apple.CoreSimulator.SimulatorTrampoline'; don't know what that is.
Simulator not installed. Please visit https://developer.apple.com/xcode/download/ to download Xcode and the iOS simulator. If you already have the latest version of Xcode installed, you may have to run the command `sudo xcode-select -s /Applications/Xcode.app`.

**Comment by fozzarelo** [comment]
Same issue.

**Comment by nahtnam** [comment]
Same here, ran the given command as well

**Comment by nahtnam** [comment]
Running `open -a Simulator` and then `npm start` fixed the issue.

**Comment by kubapyla** [comment]
Same issue, @nahtnam solution with opening simulator first from CLI is also not working, it gives me `Operation timed out`

**Comment by cpojer** [comment]
Please note this issue tracker is not a help forum (as stated in issue template).

We recommend using [StackOverflow](https://stackoverflow.com/questions/ask) where there are community incentives to help others or our [Discord channel](https://discord.gg/Xs5Xbjr) for questions.

**Comment by fozzarelo** [comment]
@cpojer there is a recent StackOverflow issue related. Only 47 views so far, but growing: 
https://stackoverflow.com/questions/55202064/expo-react-native-ios-simulator-not-working
I'll post any solution/workaround I find in there. This looks like a proper bug.

**Comment by cmackenz** [comment]
I'm having the same issue after upgrading expo-cli to `2.13.0`. 
Downgrading to `2.11.9` did not resolve the issue.

**Comment by fozzarelo** [comment]
https://stackoverflow.com/questions/55202064/expo-react-native-ios-simulator-not-working slowly getting traction! 
doubled its views in 3hours...

**Comment by nahtnam** [comment]
@cpojer I think this should be re-opened. We aren't asking for help, rather we are trying to report a bug.

**Comment by prigoose** [comment]
@nahtnam's solution opens the simulator temporarily, but the next time you open it, you have to run both commands again as well. For now, I am aliasing `start` to `open -a Simulator; expo start`.

However, I hope a better solution will be found

**Comment by GuillaumeRZ** [comment]
Hey ! Got the same issue right here...

**Comment by robbiemu** [comment]
Rather than simply looking at this as a bug to be fixed, the community desperately needs better documentation of how this is designed to work.

**Comment by fozzarelo** [comment]
Trying to narrow it down...
I think It’s a bug. 🐜
Is this related to expo?
Everyone after using expo?

**Comment by robbiemu** [comment]
I've noticed that Simulator is not gracefully exiting after closing. This is true whether or not I even run expo while it is open.

If I kill all the tasks named "Simulator" (which for some reason, cant use killall even thought they ar running as my user, have to `kill -9`those PIDs), then expo returns to working normally .. until I close the Simulator again.

**Comment by robrecord** [comment]
I noticed XCode was updated recently. Could it be related to this? Having this problem using version 10.2 (SimulatorApp-880.5 CoreSimulator-587.35)

**Comment by robbiemu** [comment]
Just a followup, I've since upgraded to the 10.14.4 release, of course rebooted, and continued my development - I now can use `killall`. I have to do this with some frequency: nearly every time I introduce a coding error that halts recompile (like unmatched parenthesis, etc), pressing 'r' in the console running expo does not reiniitialize the build. The Simulator gets stuck with a red screen that won't dismiss. if instead I relaunch the. app in ios, it goes to a blue screen but no rebuild occurs.

I am finding the best route to continue is to have two terminal tabs open, with these commands on tap:

* `expo r -c`
* `killall Simulator && open -a Simulator`

**Comment by robrecord** [comment]
Just to add, in addition to experiencing this issue using expo, I also experience an issue when using the `react-native` command:

```
Found Xcode project myAwesomeProject.xcodeproj

Could not find iPhone X simulator

Error: Could not find iPhone X simulator
    at resolve (/Users/username/Apps/myAwesomeProject/node_modules/react-native/local-cli/runIOS/runIOS.js:149:13)
    at new Promise (<anonymous>)
    at runOnSimulator (/Users/username/Apps/myAwesomeProject/node_modules/react-native/local-cli/runIOS/runIOS.js:134:10)
    at Object.runIOS [as func] (/Users/username/Apps/myAwesomeProject/node_modules/react-native/local-cli/runIOS/runIOS.js:106:12)
    at Promise.resolve.then (/Users/username/Apps/myAwesomeProject/node_modules/react-native/local-cli/cliEntry.js:117:22)
```

Also reported here: https://stackoverflow.com/questions/54504076/react-native-run-ios-returns-error-could-not-find-iphone-x-simulator

**Comment by King-of-Spades** [comment]
@robrecord Your notes remind me of symptoms I saw on an entirely different framework (Xamarin.UITest) that were caused by the update between Xcode 10.1 -> 10.2 modifying how the simulator had to be referenced in order to be connected to for debugging. And actually I've seen that connection requiring updates in the framework many times in the past, though the symptoms were usually a little less extreme then in 10.1 -> 10.2. 

Even though I'm not super familiar with React Native yet nor the raw data you shared (I stumbled here while on merely vaguely related research); it does make me wonder if React Native (and maybe other non-Apple tools) are sensitive to breaking with certain iOS simulator updates that are applied in Xcode's release cycle. 

Not enough to state anything conclusive or that interesting, but I'll definitely be keeping an eye out for such symptoms when future Xcode versions come out.

**Comment by mkempinsky** [comment]
I was able to resolve the issue by doing the following: 

run `sudo xcode-select -s /Applications/Xcode.app`

I then got an error saying `xcode-select: error: invalid developer directory '/Applications/Xcode.app'`

To fix this, I had to rename my Xcode application from Xcode-8.2.0 (or whatever it was called) to just Xcode

Then I reopened the simulator, ran npm start, clicked 'Run on ios Simulator' in the expo client and it worked

**Comment by dichotomous** [comment]
For anyone with this problem, update your react-native version to the latest in your package.json. It will prompt you also update the necessary react package, citing a version number. At the time of this comment, I updated a repository from using react-native 0.50.3 and react 16.0.0 to react-native 0.60.3 and react 16.8.6, and everything worked fine.

**Comment by hossamnasser938** [comment]
The simplest solution for me was to:
1. hit `xcrun simctl list --json devices to list available simulators`  
2. pick one of the devices name there.
3. and force use that name on run through `react-native run-ios --simulator=[picked_one]`

**Comment by ychen183** [comment]
🎉 **I've been facing the same problem for a month, however I finally found a solution that worked for me:**
 
1. Updated my Xcode to Version 11.1 ✨
2. Opened Xcode 11.1: (1) Double checked if "Xcode 11.1" was selected as my "Command Line Tool"; (2) Under the "Components" tab, I manually downloaded a simulator (iOS 12.2)
3. Updated Expo Client to @3.1.2 ✨
```sudo npm install --unsafe-perm -g expo-cli```
4. Updated Node.js from version 11.6.0  to 12.0.0
```
sudo npm install -g n
npm cache clean -f
```
5. Restarted my project & my simulator
6. ```npm start ```

✨ Which successfully launched my iOS simulator and built my project! ✨
